From a14799aa889cbe994d7fb806e7c05990bea8a9d0 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 23 Sep 2024 16:03:15 +0800
Subject: [PATCH] cmake: Fix configure & compile errors

1/ Remove hard-coded x86 march flags.
2/ Add APIDIR for mupen64plus API dir.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 CMakeLists.txt | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d36f4be..36c8c80 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,6 @@ project(parallel-rsp LANGUAGES CXX C)
 
 set(NAME_PLUGIN_M64P "mupen64plus-rsp-parallel")
 
-include_directories(../mupen64plus-core/src/api)
 add_definitions(-DM64P_PLUGIN_API)
 add_definitions(-DPARALLEL_INTEGRATION)
 
@@ -73,10 +72,12 @@ add_library(${NAME_PLUGIN_M64P} SHARED
 		arch/simd/rsp/vmulm.h)
 
 set_target_properties(${NAME_PLUGIN_M64P} PROPERTIES PREFIX "")
+option(APIDIR "Mupen64plus API dir." "../mupen64plus-core/src/api")
+target_include_directories(${NAME_PLUGIN_M64P} PUBLIC ${APIDIR})
 target_include_directories(${NAME_PLUGIN_M64P} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(${NAME_PLUGIN_M64P} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/arch/simd/rsp)
 target_include_directories(${NAME_PLUGIN_M64P} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/api)
-target_compile_options(${NAME_PLUGIN_M64P} PRIVATE ${PARALLEL_RSP_CXX_FLAGS} -march=x86-64-v3)
+target_compile_options(${NAME_PLUGIN_M64P} PRIVATE ${PARALLEL_RSP_CXX_FLAGS})
 
 option(PARALLEL_RSP_BAKED_LIGHTNING "Use built-in Lightning." ON)
 
@@ -97,8 +98,5 @@ if (PARALLEL_RSP_BAKED_LIGHTNING)
 		target_sources(lightning PRIVATE win32/mman/sys/mman.c)
 		target_include_directories(lightning PRIVATE win32/mman)
 	endif()
-	target_compile_options(lightning PRIVATE -march=x86-64-v3)
-	target_link_options(lightning PRIVATE -march=x86-64-v3)
 endif()
-target_link_options(${NAME_PLUGIN_M64P} PRIVATE -march=x86-64-v3)
 target_link_libraries(${NAME_PLUGIN_M64P} PUBLIC lightning)
-- 
2.20.1

